TheAlienThatEatsTheCarrot/GameEngine/Systems/GameSystems/MovementSystem.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  MovementSystems.swift
+//  TheAlienThatEatsTheCarrot
+//
+//  Created by Justin Cheah Yun Fei on 15/3/24.
+//
+
+import Foundation
+
+class MovementSystem: System {
+    var nexus: Nexus
+    
+    init(nexus: Nexus) {
+        self.nexus = nexus
+    }
+    
+    func update(deltaTime: CGFloat) {
+        let movableComponents = nexus.getComponents(of: MovableComponent.self)
+        for movableComponent in movableComponents {
+            let entity = movableComponent.entity
+            guard let positionalComponent = nexus.getComponent(of: PositionalComponent.self, for: entity) else {
+                return
+            }
+            let newPosition = calculateNewPosition(
+                currentPosition: positionalComponent.position,
+                direction: movableComponent.direction,
+                length: movableComponent.length,
+                deltaTime: deltaTime
+            )
+            
+            positionalComponent.position = newPosition
+            print("New position: x \(newPosition.x) y \(newPosition.y)")
```

Comment:
I think it may be better to keep such debug statements as comments when pushing the code

Reply:
will remove those!